### DISCUSSION TITLE
Correct types for form to import into reusable component?

### DISCUSSION THREAD

**Discussion by MariuzM** [TanStack/form#1019] [Q&A] [answered]
I'm trying to build reusable Input component and just having form.Field there, but struggling with passing correct types to form prop 

<img width="1004" alt="image" src="https://github.com/user-attachments/assets/d5f6f7be-c963-4ad3-8a46-c797487db0fd">

**Comment by Balastrong** [comment, marked as answer]
This draft might help you: https://github.com/TanStack/form/pull/825/files#diff-b2e0803ded7363bf3432f160f7c7f6e9bf4f982ad13b08ecb4789c228774f956R22

**Reply by MariuzM** [reply]
Thank you this is what i needed!

**Reply by shinobu-uwu** [reply]
I know that this have been answer but when I try to import `ReactFormApi` it says `Module '"@tanstack/react-form"' has no exported member 'ReactFormApi'. [2305]`, do you know why?

**Reply by shinobu-uwu** [reply]
Nevermind, I found out I can do this to get the correct typing

```ts
function MyComponent({ form }: { form: ReturnType<typeof useForm<MyValues>> }) {
    return (/* ...*/);
}

```

**Reply by Balastrong** [reply]
What version of form do you have installed? The type should be available

**Reply by shinobu-uwu** [reply]
`^0.26.4`, I updated to `0.37.1` and now the type is available. Thank you!
